pkg/fleets/controller.go

```diff
@@ -173,39 +224,102 @@ func (c *Controller) syncFleet(key string) error {
 		return err
 	}
 
-	var activeGsSet *stablev1alpha1.GameServerSet
+	activeGsSet, rest := c.filterGameServerSetByActive(fleet, list)
```

Comment:
I want to make sure I understand this piece of code. This method returns the active gameserver set a a list of previous one. We then go on to set the replicas count to 0 on those and then deleting them. What happens if one of those replicas was allocated?

Reply:
Excellent question - any previously `Allocated` `GameServers` that part of the "inactive" `GameServerSet` are left alone, and the `GameServerSet` will stay alive until the `Allocated` `GameServer` shuts itself down or is otherwise terminated.

For reference you can see the code for scaling down `GameServerSet` [here](https://github.com/GoogleCloudPlatform/agones/blob/master/pkg/gameserversets/controller.go#L304), and how it ignores `Allocated` `GameServers`.

For a live demo, you can actually see it in action on [this twitch video](https://www.twitch.tv/videos/259103050) (only live for 13 more days, will push to YouTube at some point today).

Reply:
As an interesting note - this is why `deleteEmptyGameServerSets` ([code](https://github.com/GoogleCloudPlatform/agones/pull/199/files/980ad2dd64c2b1c28db2ae92d33cdf4aebd14c5e#diff-7c0697b565f7e02d6db88c116bb970f1R295)) only deletes a `GameServerSet` when the `gsSet.Status.Replicas == 0` - because this will never be 0 when there are still `Allocated` `GameServers`.